ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -79,10 +84,9 @@ gcp_connection::put_object(const std::string &object_key, const std::string &fil
       _gcp_client.UploadFile(file_path, _bucket_name, _object_prefix + object_key);
 
     // Check if file has been successfully uploaded.
-    if (!metadata) {
-        std::cerr << "Upload failed: " << metadata.status() << std::endl;
-        return -1;
-    }
+    if (!metadata)
+        return handle_error(metadata.status(),
+          "Upload of '" + object_key + "' failed: " + metadata.status().message());
```

Comment:
This likely applies to all your error messages: I believe you want to print out `_object_prefix + object_key` rather than just `object_key` as that's the argument you are passing to `UploadFile`.

Reply:
thanks for picking this up! 
updated: 03e045308ad62715dfdd2bccdb6f701e8d94ff83